packages/cli/src/commands/update.ts

```diff
@@ -0,0 +1,128 @@
+
+        if (triggerDevPackages.length === 0) {
+            logger.info('No @trigger.dev packages found in package.json.');
+            return;
+        }
+
+        const latestVersions = await getLatestVersions(triggerDevPackages);
+        const updates: { [packageName: string]: string } = {};
+
+        triggerDevPackages.forEach(packageName => {
+            const currentVersion = installedPackages.dependencies[packageName].version;
+            const latestVersion = latestVersions[packageName];
+            if (latestVersion) {
+                if (semver.gt(latestVersion, currentVersion)) {
+                    updates[packageName] = latestVersion;
+                }
+            }
+        });
+
+
+        if (Object.keys(updates).length === 0) {
+            console.log('No updates available.');
+            return;
+        }
+
+        const confirm = await confirmUpdate(updates);
+
+        if (confirm) {
+            const updatedPackageJson = updatePackageJson(updates);
+            await writeJSONFile('package.json', updatedPackageJson);
+            logger.info('package.json updated.');
+
+            // Run 'npm install' to install the updated packages
+            logger.info('Running npm install...');
+            await installDependencies(resolvedPath);
+            logger.info('Dependencies updated.');
+        } else {
+            logger.info('Update canceled.');
+        }
+    } catch (error) {
+        logger.error('An error occurred:', error);
+    }
+}
+
+async function getInstalledPackages(): Promise<any> {
+    const manager = getUserPkgManager();
+    return new Promise((resolve, reject) => {
+        exec(`${manager} list --depth=0 --json`, (error, stdout) => {
+            if (error) {
+                reject(error);
+            } else {
+                const installedPackages = JSON.parse(stdout.toString());
+                resolve(installedPackages);
+            }
+        });
+    });
+}
+
+async function getLatestVersions(packageNames: string[]): Promise<{ [packageName: string]: string }> {
+    const registryUrl = 'https://registry.npmjs.org';
+
+    const requests = packageNames.map(packageName =>
+        axios.get(`${registryUrl}/${packageName}`, {
+                headers: {
+                    'Accept-Encoding': 'application/json',
+                }
+            }
+        )
+    );
+
+    const responses = await Promise.all(requests);
+    return responses.reduce((versions: { [packageName: string]: string }, response, index) => {
+        const packageName = packageNames[index];
+        if (packageName) {
+            const nextVersion = response.data['dist-tags']?.next;
+            const latestVersion = response.data['dist-tags'].latest;
+
+            if (nextVersion && semver.gt(nextVersion, latestVersion)) {
+                versions[packageName] = nextVersion;
+            } else {
+                versions[packageName] = latestVersion;
+            }
```

Comment:
This is really close, but we also need to check and make sure the existing package version is using the `next` tag. 

I think the pseudocode would be something like

```
If installed package is using the next tag
  Use the newest version of either the published next tag or the published latest tag
If installed package is using the latest tag
  use the newest version of only the published latest tag
```

 
We never want to go from `latest` to `next`, but if `latest` is newer than `next`, we should go from `next` to `latest`.

Reply:
so if I understand correctly I should check in the package.json and if I see "@trigger.dev/*":"next", I should updated to the most recent version even if its latest, otherwise use the latest. Is that correct?

Reply:
Yea to see if the installed version is tagged with `next` you would match against the version string for the string `"next"`, probably just with `const isNext = version.includes("next")` or something similar

Reply:
@omernave does what I said above make sense? Keen to get this PR merged as it's so close and it'd be a really great addition

Reply:
@ericallam updated the pr based on the last comment, can you take a look?